File: Zajecia1.cs
using System;
using System.Globalization;

class Program
{
    static void Main()
    {
        CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;

        while (true)
        {
            Console.WriteLine("\nMENU GLOWNE");
            Console.WriteLine("1. Zadanie 1: Prosty kalkulator dwoch liczb");
            Console.WriteLine("2. Zadanie 2: Konwerter temperatur (Celsjusz <-> Fahrenheit)");
            Console.WriteLine("3. Zadanie 3: Srednia ocen ucznia");
            Console.WriteLine("0. Wyjscie z programu");
            Console.Write("Wybierz zadanie (0-3): ");

            string choice = Console.ReadLine();
            switch (choice)
            {
                case "1":
                    RunKalkulator();
                    break;
                case "2":
                    RunTempKonwerter();
                    break;
                case "3":
                    RunSredniaOcen();
                    break;
                case "0":
                    Console.WriteLine("Zamykanie programu.");
                    return;
                default:
                    Console.WriteLine("Nieprawidlowy wybor. Sprobuj ponownie.");
                    break;
            }
        }
    }

    static void RunKalkulator()
    {
        Console.Write("Podaj pierwsza liczbe: ");
        double a = double.Parse(Console.ReadLine());

        Console.Write("Podaj druga liczbe: ");
        double b = double.Parse(Console.ReadLine());

        Console.Write("Wybierz operacje (+, -, *, /): ");
        string op = Console.ReadLine();

        if (op == "+")
        {
            Console.WriteLine($"Wynik: {a + b}");
        }
        else if (op == "-")
        {
            Console.WriteLine($"Wynik: {a - b}");
        }
        else if (op == "*")
        {
            Console.WriteLine($"Wynik: {a * b}");
        }
        else if (op == "/")
        {
            if (b == 0)
                Console.WriteLine("Blad: Nie można dzielic przez zero.");
            else
                Console.WriteLine($"Wynik: {a / b}");
        }
        else
        {
            Console.WriteLine("Nieznana operacja arytmetyczna.");
        }
    }

    static void RunTempKonwerter()
    {
        Console.Write("Wybierz kierunek (C - Celsjusz na Fahr, F - Fahr na Celsjusz): ");
		string direction = Console.ReadLine().Trim().ToUpper();

        Console.Write("Podaj wartosc temperatury: ");
        string inputTemp = Console.ReadLine();

		if (!double.TryParse(inputTemp, out double temp))
		{
			Console.WriteLine("Blad: Wprowadzono niepoprawny format liczby.");
			return;
		}
		if (direction == "C")
		{
			double fahrenheit = temp * 1.8 + 32;
			Console.WriteLine($"{temp} C = {fahrenheit} F");
		}
		else if (direction == "F")
		{
			double celsius = (temp - 32) / 1.8;
			Console.WriteLine($"{temp} F = {celsius} C");
		}
        else
        {
            Console.WriteLine("Nieprawidlowy kierunek konwersji.");
        }
    }

    static void RunSredniaOcen()
    {
        Console.Write("Podaj liczbe ocen: ");
        int count = int.Parse(Console.ReadLine());
        
        if (count <= 0)
        {
            Console.WriteLine("Liczba ocen musi byc wieksza od 0.");
            return;
        }

        double sum = 0;
        for (int i = 0; i < count; i++)
        {
            Console.Write($"Podaj ocene {i + 1} (1-6): ");
            sum += double.Parse(Console.ReadLine());
        }

        double average = sum / count;
        Console.WriteLine($"Srednia: {average:F2}");

        if (average >= 3.0)
            Console.WriteLine("Uczen zdal.");
        else
            Console.WriteLine("Uczen nie zdal.");
    }
}

File: zajecia1.py
import sys

def run_kalkulator():
    try:
        a = float(input("Podaj pierwszą liczbę: "))
        b = float(input("Podaj drugą liczbę: "))
        op = input("Wybierz operację (+, -, *, /): ").strip()

        if op == "+":
            print(f"Wynik: {a + b}")
        elif op == "-":
            print(f"Wynik: {a - b}")
        elif op == "*":
            print(f"Wynik: {a * b}")
        elif op == "/":
            if b == 0:
                print("Błąd: Nie można dzielić przez zero.")
            else:
                print(f"Wynik: {a / b}")
        else:
            print("Nieznana operacja.")
    except ValueError:
        print("Błąd: Wprowadzono niepoprawny format liczby.")

def run_konwerter_temperatur():
    try:
        direction = input("Wybierz kierunek (C – Celsjusz na Fahr, F – Fahr na Celsjusz): ").strip().upper()
        temp = float(input("Podaj wartość temperatury: "))

        if direction == "C":
            fahrenheit = temp * 1.8 + 32
            print(f"{temp}°C = {fahrenheit}°F")
        elif direction == "F":
            celsius = (temp - 32) / 1.8
            print(f"{temp}°F = {celsius}°C")
        else:
            print("Nieprawidłowy kierunek konwersji.")
    except ValueError:
        print("Błąd: Wprowadzono niepoprawny format liczby.")

def run_srednia_ocena():
    try:
        count = int(input("Podaj liczbę ocen: "))
        if count <= 0:
            print("Liczba ocen musi być większa od 0.")
            return
    except ValueError:
        print("Błąd: Liczba ocen musi być liczbą całkowitą.")
        return

    total_sum = 0
    for i in range(count):
        while True:
            try:
                grade = float(input(f"Podaj ocenę {i + 1} (1-6): "))
                
                if 1 <= grade <= 6:
                    total_sum += grade
                    break
                else:
                    print("Błąd: Ocena musi mieścić się w przedziale od 1 do 6.")
                    
            except ValueError:
                print("Błąd: Wprowadzono niepoprawny format liczby. Spróbuj ponownie.")

    average = total_sum / count
    print(f"Średnia: {average:.2f}")

    if average >= 3.0:
        print("Uczeń zdał.")
    else:
        print("Uczeń nie zdał.")

def main():
    while True:
        print("\nMENU GŁÓWNE")
        print("1. Zadanie 1: Prosty kalkulator dwóch liczb")
        print("2. Zadanie 2: Konwerter temperatur (Celsjusz ↔ Fahrenheit)")
        print("3. Zadanie 3: Średnia ocen ucznia")
        print("0. Wyjście z programu")
        choice = input("Wybierz podzadanie (0-3): ").strip()

        if choice == "1":
            run_kalkulator()
        elif choice == "2":
            run_konwerter_temperatur()
        elif choice == "3":
            run_srednia_ocena()
        elif choice == "0":
            print("Zamykanie programu.")
            sys.exit(0)
        else:
            print("Nieprawidłowy wybór. Spróbuj ponownie.")

if __name__ == "__main__":
    main()